acdcli/api/content.py

```diff
@@ -61,57 +62,63 @@ class ContentMixin(object):
     """Implements content portion of the ACD API."""
 
     def create_folder(self, name: str, parent=None) -> dict:
-        body = {'kind': 'FOLDER', 'name': name}
-        if parent:
-            body['parents'] = [parent]
-        body_str = json.dumps(body)
+        while True:
+            body = {'kind': 'FOLDER', 'name': name}
+            if parent:
+                body['parents'] = [parent]
+            body_str = json.dumps(body)
 
-        acc_codes = [http.CREATED]
+            acc_codes = [http.CREATED]
 
-        r = self.BOReq.post(self.metadata_url + 'nodes', acc_codes=acc_codes, data=body_str)
+            r = self.BOReq.post(self.metadata_url + 'nodes', acc_codes=acc_codes, data=body_str)
+            if r.status_code in RETRY_CODES: continue  # the fault lies not in our stars, but in amazon
```

Comment:
In practice, Amazon kicks the client well before any infinite loops happen, usually with an error in the 400 range that we're deliberately not catching. I've seen errors in the family of requests/second, unable to process, and sometimes just unavailable when amazon goes down for short periods and needs to kick clients. We're not catching those on purpose!

Otherwise, for file system stability the goal here is to reduce errors we'd need to propagate upwards to the client.

If you have an example of a non-ending loop, please send me the response codes involved and I'll change the set we retry over.